src/tplink_omada_client/omadasiteclient.py

```diff
@@ -455,3 +455,19 @@ async def get_gateway(self, mac_or_device: Union[str, OmadaDevice, None]) -> Oma
         )
 
         return OmadaGateway(result)
+
+    async def set_led_setting(self, mac_or_device: Union[str, OmadaDevice, None], setting: LedSetting) -> bool:
+        """Sets the onboard LED setting for the device"""
+        if isinstance(mac_or_device, OmadaDevice):
+            mac = mac_or_device.mac
+        else:
+            mac = mac_or_device
+
+        payload = {"mac": mac, "ledSetting": setting.value }
+        await self._api.request(
+            "patch",
+            self._api.format_url(f"eaps/{mac}", self._site_id),
```

Comment:
There is a different URL for setting the device LED setting on Gateways and Switches too. It would be worth supporting all three.